RationaleMCP/0031/concept.md

```diff
@@ -0,0 +1,135 @@
+
+### Types
+
+* Following types exist:
+   * predefined types: `Real`, `Integer`, `Boolean`, `String`, `ExternalObject`,
+   * record types
+   * functions
+   * simple types.
+   
+* Properties of record definitions are:
+   * The components are defined as variable definitions. Declaration equations are not allowed.
+   * There are no equations or algorithms.
+   * record definitions can not nest.
+   * There is no automatism to generate record constructors anymore. They have to be made explicit as functions.
+
+* Simple types are aliases to predefined types enriched with annotations. 
+   * They are used to avoid duplication of annotations in variable definitions. 
+   * An annotation of a variable definition overrides the same annotation of the type.
+   * They have no dimension.
+
+* The order of definition is not important.
+
+### Variables
+
+* A variable definition consists of
+   1. a prefix (e.g. `input`,`output`),
+   1. a type name,
+   1. a variable name
+   1. an optional dimension specification,
+   1. an optional declaration equations,
+   1. an optional description string and
+   1. an optional annotation.
+* The dimension of an array is defined by a list of dimension sizes inside square brackets, e.g. `[2,3]`. The dimension sizes are allowed to be: 
+   * integer literals,
+   * `:`. In this case is determined from the dimension of the right side of the declaration equation, if exist and if it is possible. If not, the size is seen as unknown and no subscription is possible.
+   * Only in functions: expressions containing literals and references to input parameters and other (predefined) pure functions
+* Parameters and Constants from Modelica are mapped to variables with special annotations.
+* It is assumed, that constants and parameters of the Modelica model was evaluated already evaluated if needed, i.e. because dimensions or branches or loops in equations depend from them.
+* fixed start attributes are converted to initial equations, non fixed start attributes to annotations
+
+### Expressions
+
+* Expressions are defined as in Modelica.
+* There are the predefined operators and functions of Modelica and some more to support runtime behavior, e.g. event handling
+* There is no implied behavior of relations and builtin operators with regard to event handling any more. All has to be made explicit by introduction of helper variables and equations.
+* The operators `pre(...)`, `previous(...)`, `sample(...)`, `delay(...)`, `spatialDistribution(...)` have the same semantics as in Modelica.
+* Function calls have no default arguments, all must be explicitly specified in the correct order. There are no named arguments.
+
+### Equations and Algorithms 
+* Equation and algorithm sections are defined as in Modelica, including initial equations and algorithms.
+* extension: equation and algorithm sections may have a dimension (to retain the information about vectorized components).
```

Comment:
Could you elaborate the idea about dimensions?

Reply:
Will do later on.